src/main/java/world/bentobox/bentobox/listeners/flags/protection/EntityInteractListener.java

```diff
@@ -38,7 +40,9 @@ public void onPlayerInteractEntity(PlayerInteractEntityEvent e) {
                 checkIsland(e, e.getPlayer(), e.getRightClicked().getLocation(), Flags.RIDING);
             }
             // Minecart riding
-            else if (e.getRightClicked() instanceof Minecart) {
+            else if (e.getRightClicked() instanceof RideableMinecart
+				|| e.getRightClicked() instanceof CommandMinecart
+				|| e.getRightClicked() instanceof ExplosiveMinecart) {
```

Comment:
I'd rather have used an exclusion here - because maybe Mojang will add new Minecarts at some point, who knows?

Replace this with the following:
```java
else if (e.getRightClicked() instanceof Minecart &&
          (! (e.getRightClicked() instanceof FurnaceMinecart || e.getRightClicked() instanceof HopperMinecart || e.getRightClicked() instanceof ChestMinecart) ) {
    // code
}
```

Reply:
I do not agree with this
I added CommandMinecart and ExplosiveMinecart here since they were not processed elsewhere, just in case.
I did not find flag that prevents on TNT explosions, so I would like Explosive Minecart to be protected by TNT explosion flag, and in this flag left only RidableMinecart. 
As Minecart flag basically is about users ability to ride minecart, not to use minecart object. Thats my thoughts.

Reply:
I'm testing with the RideableMinecart only. Please open an issue for the TNT Minecarts, I'm afraid they will need some special handling - but this can wait another release.